armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -57,118 +69,191 @@ def _determineLinked(componentA, componentB):
                 "they are going to be assumed to not be linked.",
                 single=True,
             )
-            linked = False
+            return False
+        # Check if one component could fit within the other
+        idA = componentA.getCircleInnerDiameter(cold=True)
+        odA = componentA.getBoundingCircleOuterDiameter(cold=True)
+        idB = componentB.getCircleInnerDiameter(cold=True)
+        odB = componentB.getBoundingCircleOuterDiameter(cold=True)
+        biggerID = max(idA, idB)
+        smallerOD = min(odA, odB)
+        return biggerID < smallerOD
+    return False
+
+
+# Make a generic type so we can "template" the axial link class based on what could be above/below a thing
+Comp = typing.TypeVar("Comp", Block, Component)
+
+
+@dataclasses.dataclass
+class AxialLink(typing.Generic[Comp]):
+    """Small class for named references to objects above and below a specific object.
+
+    Axial expansion in ARMI works by identifying what objects occupy the same space axially.
+    For components, what components in the blocks above and below axially align? This is used
```

Comment:
```suggestion
    Axial expansion in ARMI works by identifying what objects occupy the same axial space.
    For components in blocks, identify which above and below axially align. This is used
```

Reply:
Something struck me as off about the phrasing, but I'm not married to my suggestion. Take it or leave it.